### PR TITLE
Added db190 keyboard

### PR DESCRIPTION
**Pull request by itarze**
## Description

This keyboard has been verified to work properly.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/westberry/db190/info.json [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "haptic": true,
+        "encoder": true,
+        "rgb_matrix": true,
+        "rgblight": false,
+        "magic_keycode": true
```
Comment (drashna) [inline keyboards/westberry/db190/info.json:28]:
```suggestion
```

File: keyboards/westberry/db190/info.json [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    },
+    "bootmagic": {
+        "matrix": [0, 0]
+    },
```
Comment (drashna) [inline keyboards/westberry/db190/info.json:32]: This is default behavior, and shouldn't be needed

File: keyboards/westberry/db190/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+
+# WestBerryTech DB190 keyboard
+
+
+*A short description of the keyboard/project*
```
Comment (drashna) [inline keyboards/westberry/db190/readme.md:5]: ?

File: keyboards/westberry/db190/rules.mk [hunk 4]
```diff
@@ -0,0 +1,6 @@
+RGB_MATRIX_CUSTOM_USER = yes # Enable custom RGB effect
```
Comment (drashna) [inline keyboards/westberry/db190/rules.mk:1]: is at the keyboard level, should only enable kb version.
```suggestion
RGB_MATRIX_CUSTOM_KB = yes
```

File: keyboards/westberry/db190/rules.mk [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+EEPROM_DRIVER = wear_leveling
+WEAR_LEVELING_DRIVER = embedded_flash
```
Comment (drashna) [inline keyboards/westberry/db190/rules.mk:3]: This shouldn't be needed, as this is the default behavior, already. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by itarze** [comment]
Can merge?

**Comment by itarze** [comment]
Can merge?

**Comment by itarze** [comment]
Hi, bro. Can merge?

File: keyboards/westberry/db190/db190.c [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+   {RS_ROW12, RS_COL3 }, // 77 R_Ctrl
+   {RS_ROW13, RS_COL3 }, // 78 LEFT
+   {RS_ROW14, RS_COL3 }, // 79 DOWN
+   {RS_ROW16, RS_COL2 }  // 80 RIGHT
+};
+
+// clang-format on
+#endif
+
+void matrix_init_kb(void) {
```
Comment (drashna) [inline keyboards/westberry/db190/db190.c:114]: This should be:
```suggestion
void keyboard_pre_init_kb(void) {
```

(or possibly `keyboard_post_init_kb`, if there are issues with init-ing too early.

File: keyboards/westberry/db190/db190.c [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+#ifdef BOOTMAGIC_LITE
+    rgb_driver_init();
+#endif
```
Comment (drashna) [inline keyboards/westberry/db190/db190.c:117]: This is init'ed with the normal rgb matrix init.  Why is this needed here?   Especially only if bootmagic is enabled?

File: keyboards/westberry/db190/rgb_matrix_kb.inc
Comment (drashna) [inline keyboards/westberry/db190/rgb_matrix_kb.inc:1]: This isn't really needed.  Turning off rgb matrix, or setting the flags to "none" will both accomplish this, without the need for a separate mode.  Also, setting the mode to solid color, and the val to 0 will also accomplish this, as well.  And both without the need to add a separate mode.

File: keyboards/westberry/common/readme.md [hunk 8]
```diff
@@ -0,0 +1,8 @@
+## Description
+On a limited hardware budget, you can use WB32's PWM Channels + some mos tubes to drive your rgb
```
Comment (drashna) [inline keyboards/westberry/common/readme.md:2]: So this is using a bunch of pins to drive a bunch of dumb rgb leds? 

Namely, more information about what this is doing and why would be very useful here.

File: keyboards/westberry/common/wb32_driver.c [hunk 9]
```diff
@@ -0,0 +1,526 @@
...
+    wb32_pwm_init();
+    wb32_pwm_pin_init();
+    // Start TIM1.
+    TIM1->CR1 |= TIM_CR1_CEN;
+    wbrgb_set_color_all(0x00, 0x00, 0x00);
+  }
+}
+
+
+/* The following is Key Matrix related. */
```
Comment (drashna) [inline keyboards/westberry/common/wb32_driver.c:421]: why is there switch matrix and rgb matrix code here?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/westberry/db190/db190.c (hunk 7 above)
Comment (itarze) [reply keyboards/westberry/db190/db190.c:117]: Matrix Cols are reused in RGB SW Pins due to insufficient IO quantity. Therefore, if bootmagic is used, PWM Pins also needs to be configured, otherwise, I/O level cannot be correctly detected. This is caused by insufficient hardware resources.

File: keyboards/westberry/common/wb32_driver.c (hunk 9 above)
Comment (itarze) [reply keyboards/westberry/common/wb32_driver.c:421]: This reply is the same as above, matrix cols are multiplexed with RGB SW Pin.

File: keyboards/westberry/db190/rules.mk [hunk 10]
```diff
@@ -0,0 +1,4 @@
+RGB_MATRIX_CUSTOM_KB = yes
+
```
Comment (itarze) [inline keyboards/westberry/db190/rules.mk:2]:
```suggestion
```

File: keyboards/westberry/db190/db190.c [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+};
+
+// clang-format on
+#endif
+
+void keyboard_pre_init_kb(void) {
+#ifdef BOOTMAGIC_LITE
+    rgb_driver_init();
+#endif
+    matrix_init_user();
```
Comment (itarze) [inline keyboards/westberry/db190/db190.c:118]:
```suggestion
    keyboard_pre_init_user();
```

**Comment by itarze** [comment]
@tzarc Under what circumstances should it be marked as invalid label? Is this PR not going to be merged?
